src/main/kotlin/com/ktor/solution/Application.kt

```
@@ -0,0 +1,202 @@
+package com.ktor.solution
+
+import com.ktor.solution.Config.API_BASE_URL
+import io.ktor.server.engine.embeddedServer
+import io.ktor.server.netty.Netty
+
+import io.ktor.server.application.*
+import io.ktor.server.response.*
+import io.ktor.server.routing.*
+import io.ktor.client.plugins.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.client.HttpClient
+import io.ktor.client.call.*
+
+import io.ktor.server.request.*
+import io.ktor.http.*
+
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation as ClientContentNegotiation
+import io.ktor.server.plugins.contentnegotiation.ContentNegotiation as ServerContentNegotiation
+import io.ktor.serialization.kotlinx.json.*
+import kotlinx.serialization.json.Json
+
+import io.ktor.client.plugins.defaultRequest
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class Post(val userId: Int, val id: Int, val title: String, val body: String)
+
+@Serializable
+data class Comment(val postId: Int, val id: Int, val name: String, val email: String, val body: String)
+
+object Config {
+    const val API_BASE_URL = "https://jsonplaceholder.typicode.com"
+}
+
+fun main() {
+
+    val client = HttpClient {
+        install(ClientContentNegotiation) {
+            json(Json {
+                encodeDefaults = true
+                isLenient = true
+                coerceInputValues = true
+                ignoreUnknownKeys = true
+            })
+        }
+        defaultRequest {
+            host = API_BASE_URL
+            port = 80
+        }
+    }
+
+    suspend fun updatePost(id: Int, newPostData: Post): Post? {
+        return try {
+            client.put("${API_BASE_URL}/posts/$id") {
+                contentType(ContentType.Application.Json)
+                setBody(newPostData)
+            }.body<Post>()
+        } catch (e: Exception) {
```

Do we need this try-catch with swallowed exceptions here?